packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart

```diff
@@ -193,6 +194,52 @@ void main() {
       expect(defaultPath, api.returnSelectedElement(hResult, dialog));
       verify(mockFileOpenDialogAPI.getUserSelectedPath(any)).called(1);
     });
+
+    test('setInitialDirectory should return param if initialDirectory is empty',
+        () {
+      expect(defaultReturnValue,
+          api.setInitialDirectory(defaultReturnValue, '', dialog));
+    });
+
+    test(
+        'setInitialDirectory should return same hResult if initialDirectory is null',
+        () {
+      const int anyHResult = 999;
+
+      expect(anyHResult, api.setInitialDirectory(anyHResult, null, dialog));
+    });
+
+    test(
+        'setInitialDirectory should return same hResult if initialDirectory is null',
+        () {
+      const int anyHResult = 999;
+
+      expect(anyHResult, api.setInitialDirectory(anyHResult, null, dialog));
+    });
+
+    test('setInitialDirectory should success when initialDirectory is valid',
+        () {
+      expect(successReturnValue,
+          api.setInitialDirectory(defaultReturnValue, defaultPath, dialog));
+    });
+
+    test('setInitialDirectory should throw Error 0x80070002 when initialDirectory is an inexistent path',
+        () {
+      expect(
+          () => api.setInitialDirectory(defaultReturnValue, r'C:\INEXISTENT_DIR', dialog),
```

Comment:
Can we guarantee that a path is available? What if I run this test suite on Linux?

Reply:
Not really, wouldn't that be like running mac tests on windows?

Reply:
jajaja is true, but what about if `c:` is unavailable? My concern is related to the CI sever running these tests.

Reply:
Maybe we can go with just 'INEXISTENT_DIR' to be sure 🤔, I'll test how it runs in windows

Reply:
just tested it and it runs smoothly, so I'll change it to that